src/actions/accountActions.js

```diff
@@ -83,6 +83,24 @@ export function updateAccountStats() {
 	}
 }
 
+export function updateAccountSettings() {
+	return async function(dispatch, getState) {
+		const { identity } = getState().persist.account
+		try {
+			const settings = {}
+			settings.grantType = await getGrantType({ identity: identity.address })
```

Comment:
make a little optimization here or where you call this function.
Call for the `grantType` only if it is `undefined`. `null` is if there is no grant for the account.